code/game/objects/items/explosives/grenades/marines.dm

```diff
@@ -364,3 +364,56 @@
 /obj/item/explosive/grenade/flare/cas/turn_off()
 	QDEL_NULL(target)
 	return ..()
+
+/obj/item/explosive/grenade/minelayer
+	name = "\improper M21 APRDS \"Minelayer\""
+	desc = "Anti-Personnel Rapid Deploy System, APRDS for short, is a device designed to quickly deploy M20 mines in large quantities. This variant deploys 5 mines. WARNING: Operating in tight places or existing mine fields will result in reducing the amount of placed mines."
+	icon_state = "grenade_hefa"
+	item_state = "grenade_hefa"
+	icon_state_mini = "grenade_green"
+	throw_range = 2
+	det_time = 10 SECONDS
+	///radius on mine placement
+	var/range = 3
+	///amount of mines that we attempt to place
+	var/amount = 5
+	///amount of mines placed
+	var/placed = 0
+	///who activated the grenade
+	var/mob/user = null //we need user to get iff signal
+
+/obj/item/explosive/grenade/minelayer/attack_self(mob/M)
+	user = M
+	. = ..()
+
+/obj/item/explosive/grenade/minelayer/prime()
+	if(!user) //if chain exploded; we dont want to place mines without iff
+		explosion(loc, light_impact_range = 1, flash_range = 2)
+		qdel(src)
+		return
+
+	var/obj/item/card/id/id = user.get_idcard()
+	var/list/turf_list = list()
+
+	for(var/turf/T in view(range, loc))
+		turf_list += T
+
+	for(var/i = 1 to turf_list.len)
+		var/turf/U = pick(turf_list)
+		turf_list.Remove(U)
+		if(!U.density && !turf_block_check(src, U) && !(locate(/obj/item/explosive/mine) in range(1, U)))
+			var/obj/item/explosive/mine/placed_mine = new /obj/item/explosive/mine(U)
```

Comment:
this effect is so cheap. Maybe throw the mines at random location?

Reply:
Throw leaves room for error, which allows to stack multiple mines in 1 tile

Reply:
If you really don't want that you can sequentially throw mines. Or some mines will just not plant but that's not an issue either

Reply:
Also i don't see the point of this being a nade, rather than a device you can plant?

Reply:
If you want i can make it a device

Reply:
yes please